Sources/NIOPosix/Bootstrap.swift

```diff
@@ -326,6 +326,25 @@ public final class ServerBootstrap {
         return self.bind(unixDomainSocketPath: unixDomainSocketPath)
     }
 
+    #if canImport(Darwin) || os(Linux)
+    public func bind(to vsockAddress: VsockAddress) throws -> EventLoopFuture<Channel> {
+        func makeChannel(_ eventLoop: SelectableEventLoop, _ childEventLoopGroup: EventLoopGroup, _ enableMPTCP: Bool) throws -> ServerSocketChannel {
+            if enableMPTCP {
+                throw ChannelError.operationUnsupported
+            }
+            let socket = try ServerSocket(protocolFamily: .vsock, setNonBlocking: true)
+            try socket.bind(to: vsockAddress)
+            try socket.listen(backlog: 128)
```

Comment:
Preferentially we'd have the `Channel` be responsible for doing this: that is, instead of calling `bind` and `listen` here we'd have the `Channel` do it in `bind0`. This would use a user event to trigger the operation, as we do in NIOTS.